src/VSFe/item2/ChickenDriver.java

```diff
@@ -0,0 +1,4 @@
+package VSFe.item2;
+
+public class ChickenDriver {
```

Comment:
you may test multithreaded environment async using the code below

```java
	@Test
	public void testConstructorPattern() throws InterruptedException {
		int numberOfThreads = 500;
		ExecutorService service = Executors.newFixedThreadPool(500);
		CountDownLatch latch = new CountDownLatch(numberOfThreads);

		for (int i = 0; i < numberOfThreads; i++) {
			service.execute(() -> { 
				/* what you want to test */
                                 ...
				latch.countDown();
			});
		}
		latch.await();
	}
```

* newFixedThreadPool: create thread pool
* execute: allocate a thread from the pool
* [reference](https://hamait.tistory.com/937)